src/software/backend/robot_status.cpp

```diff
@@ -0,0 +1,82 @@
+#include "software/backend/robot_status.h"
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "software/logger/logger.h"
+
+TbotsRobotMsg convertRobotStatusToTbotsRobotMsg(RobotStatus* robot_status)
+{
+    std::map<std::string, ErrorCode> errorStringToEnumMap = {
```

Comment:
All these strings are copied from `ERROR_LT_MESSAGES` from `output/radio/mrf/messages.h` which isn't ideal (it's brittle). I don't have a better solution atm but we should probably make a ticket to consolidate error definitions and handling soon in the future.

@akhilveeraghanta @jonathanlew you guys know more about the wifi/radio/backend plans so deferring to you, just want to point it out.

Reply:
We could remove these maps, just use an enum instead, and use our `MAKE_ENUM` so that it's representable as a string? That would allow us to remove all the maps, which would be awesome.

Reply:
A few questions in regards to using `MAKE_ENUM` in `robot_status.cpp` and `robot_status_test.cpp`.
- When you say "use `MAKE_ENUM`", do you mean using `MAKE_ENUM` inside of `robot_component_status.proto`?
- In `robot_status.cpp`, we want to map the incoming ErrorCode vector of strings to corresponding enums. My concern is that the incoming error messages defined in `ERROR_LT_MESSAGES` from `output/radio/mrf/messages.h` don't match up exactly (i.e. string "dribbler motor hot" is not the same as what I assume is the `toString()` output from the ENUM `DRIBBLER_MOTOR_HOT`. I could just write a function to convert the string to the appropriate format to fix this but wanted to check.
- For `robot_status_test.cpp`, I would argue that we don't need to use `MAKE_ENUM` because not all of the messages will from `robot_message` will be converted (namely those that don't fall in the ErrorCode category). So having the unit test mock the `robot_messages` and `expected_error_code` to be returned seems to make more sense.

Reply:
Using `ErrorCode_descriptor()->FindValueByNumber(error_code)->name();` seems to work. :+1:

Reply:
Iterating over an enum using int is a code smell because we can't rely on the value being consecutive. 
Found a better solution:
```
bool Foo_Parse(const string& name, Foo* value): If name is a valid value name for this enum, assigns that value into value and returns true. Otherwise returns false. In the above example, Foo_Parse("VALUE_C", &some_foo) would return true and set some_foo to 1234. 
```
(https://developers.google.com/protocol-buffers/docs/reference/cpp-generated#enum)

Reply:
`ErrorCode_Parse` has been used to remove the map.